wolfcrypt/src/memory.c

```diff
@@ -942,3 +942,78 @@ void XFREE(void *p, void* heap, int type)
 
 #endif /* HAVE_IO_POOL */
 
+#ifdef WOLFSSL_MEMORY_TRACKING
```

Comment:
I'm not sure having WOLFSSL_MEMORY_TRACKING and the existing WOLFSSL_TRACK_MEMORY with WOLFSSL_DEBUG_MEMORY makes sense.  Can you add what's missing to the existing TRACK_MEMORY instead?  What do you think?

Reply:
I missed that WOLFSSL_STATIC_MEMORY stops the code in mem_track.h from being compiled.
Therefore the alternative is:
 - memory.h can be changed to have the heap and type as a parameter to the wolfSSL allocator functions when WOLFSSL_TRACK_MEMORY is defined,
 - the memory allocator functions in mem_track.h modified to take heap and type and
 - debug output change to include the type.
This is still less preferred as the allocators are set.

Reply:
My goal here is to avoid 2,3,4,5,6 different memory debug build options.  It's confusing for us, let alone a customer who barely knows our stuff.

Having WOLFSSL_TRACK_MEORY and WOLFSSL_MEMORY_TRACKING is even more confusing.

At least change the name, clearly document the differences and let's not use WOLFSSL_STATIC_MEMORY if it is in any way (that's a completely different feature).

Reply:
I want all the information that I'm getting from the logging calls.
No other mechanism gives me all the information.
I changed the define to WOLFSSL_MEMORY_LOG.